### PR TITLE
Replace deployment.toml file when require a config change.

### PR DESCRIPTION
**Pull request by hasinidilanka**
**Proposed Changes**

- Add particular configurations into `deployment.toml` file by replacing the file.

- Related to: #https://github.com/wso2/carbon-kernel/pull/2053

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/DBSeperationTestCase.java [hunk 1]
```diff
@@ -41,53 +40,24 @@
...
-        userMgtXml = new File(serverConfDir + File.separator + "user-mgt.xml");
-        File userMgtXmlToCopy = new File(artifactsDir + File.separator + "user-mgt-dbseperated.xml");
-
-        registryXml = new File(serverConfDir + File.separator + "registry.xml");
-        File registryXmlToCopy = new File(artifactsDir + File.separator + "registry-dbseperated.xml");
-
+        String carbonHome = Utils.getResidentCarbonHome();
+        File defaultTomlFile = getDeploymentTomlFile(carbonHome);
+        File configuredTomlFile = new File(getISResourceLocation() + File.separator + File.separator +
+                "provisioning" +File.separator+ "db_separation_config.toml");
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/DBSeperationTestCase.java]: Fix formatting

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java [hunk 2]
```diff
@@ -184,7 +179,7 @@ public void endTest() throws Exception {
 
         super.deleteSAML2WebSSOConfiguration(PORT_OFFSET_1, SECONDARY_IS_SAML_ISSUER_NAME);
         super.deleteServiceProvider(PORT_OFFSET_1, SECONDARY_IS_SERVICE_PROVIDER_NAME);
-        serverConfigurationManager.restoreToLastConfiguration(true);
+        serverConfigurationManager.restoreToLastConfiguration(false);
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: is this intentional?

File: modules/distribution/src/repository/resources/conf/default.json [hunk 3]
```diff
@@ -374,6 +347,8 @@
 
 
 
+  "identity.data_source": "jdbc/WSO2IdentityDB",
+  "data_source.skip_db_schema_creation": false,
```
[the remark this answers is not in the document]
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/default.json]: IdentityDB. Fixed with: https://github.com/wso2/product-is/pull/5158/commits/6fc7cd661abb956c87bf1c6b58b04e29aea5a7fc

File: modules/distribution/src/repository/resources/conf/default.json [hunk 4]
```diff
@@ -481,8 +456,10 @@
   "oauth.scope_validator.xacml.class": "org.wso2.carbon.identity.oauth2.validators.xacml.XACMLScopeValidator",
   "oauth.grant_type.saml_bearer.user_type": "FEDERATED",
 
-  "oauth.oidc.prompt_user_consent": "$ref{authentication.consent.prompt}",
+  "oauth.consent_prompt": "$ref{authentication.consent.prompt}",
```
[the remark this answers is not in the document]
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/default.json]: Fixed with: https://github.com/wso2/product-is/pull/5158/commits/6fc7cd661abb956c87bf1c6b58b04e29aea5a7fc

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/DBSeperationTestCase.java (hunk 1 above)
Comment (hasinidilanka) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/DBSeperationTestCase.java]: Fixed.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java (hunk 2 above)
Comment (hasinidilanka) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/ChangeACSUrlTestCase.java]: No. Fixed with: https://github.com/wso2/product-is/pull/5158/commits/6fc7cd661abb956c87bf1c6b58b04e29aea5a7fc

**Pull request merged**